mangaki/mangaki/utils/recommendations.py

```diff
@@ -159,30 +106,38 @@ def get_group_reco_algo(request, users_id=None, algo_name='als',
     encoded_work_ids = [algo.dataset.encode_work[work_id]
                         for work_id in filtered_works]
 
-    encoded_user_ids = []
-    extra_users_parameters = []
-    for user_id in users_id:
-        # TODO: also recompute parameters if ratings have changed
-        if user_id not in algo.dataset.encode_user:
-            if available_works is None:
-                available_works = set(algo.dataset.encode_work.keys())
-            df_rated_works = (pd.DataFrame(list(user_ratings[user_id].items()),
-                                           columns=['work_id', 'choice'])
-                                .query('work_id in @available_works'))
-            enc_rated_works = df_rated_works['work_id'].map(
-                algo.dataset.encode_work
-            )
-            user_rating_values = df_rated_works['choice'] \
-                .map(rating_values)
-            extra_users_parameters.append(algo.fit_single_user(
-                enc_rated_works,
-                user_rating_values
-            ))
-        else:
-            encoded_user_ids.append(algo.dataset.encode_user[user_id])
-
-    pos_of_best = algo.recommend(encoded_user_ids,
-                                 extra_users_parameters=extra_users_parameters,
+    my_mean, my_feat = algo.fit_single_user(
+        [algo.dataset.encode_work[work_id] for work_id in my_ratings.keys()],
+        [rating_values[choice] for choice in my_ratings.values()]
+    )
+
+    if algo.get_shortname().startswith('svd') and participating_other_ids:
+        he = HomomorphicEncryption(participating_other_ids, quantize_round=1,
+                                   MAX_VALUE=800 * group_length)
+        transform = he.encrypt_embeddings
+        is_encrypted = True
+    else:
+        transform = lambda _, parameters: parameters  # Return the embeddings
+        is_encrypted = False
+
+    embeddings = []
+    for user_id in participating_other_ids:
+        user_ratings = df.query("user_id == @user_id")
+        embeddings.append(transform(user_id,
+            algo.fit_single_user(
+                user_ratings['encoded_work_id'],
+                user_ratings['rating']
+            )))
+
+    if is_encrypted:
+        sum_means, sum_feats = he.decrypt_embeddings(embeddings)
+        group_parameters = [[(my_mean + sum_means) / group_length,
+                             (my_feat + sum_feats) / group_length]]
```

Comment:
This should probably be a tuple and not an array for consistency with the unencrypted case
```python
        group_parameters = [((my_mean + sum_means) / group_length,
                             (my_feat + sum_feats) / group_length)]
```